### PR TITLE
#2628 Move CHighPrecisionTimer to util

### PR DESCRIPTION
**Pull request by pljones**
**Short description of changes**

Removes dependency of Linux sound code on Server code by moving High Precision Timer to join the other timer code in Util.

CHANGELOG: Internal: code restructuring to reduce cross-dependencies (CHighPrecisionTimer)

**Context: Fixes an issue?**
Fixes: #2628

**Does this change need documentation? What needs to be documented and how?**
No.

**Status of this Pull Request**
Running on all my clients and servers.

**What is missing until this pull request can be merged?**
Nothing really but I'd wait until after 3.9.0, as moving to util might not be its final resting place.

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

File: src/sound/jack/sound.h [hunk 1]
```diff
@@ -34,15 +34,15 @@
...
-
 #if WITH_JACK
 #    include <jack/jack.h>
 #    include <jack/midiport.h>
 #endif
 
+#include "../../global.h"
+#include "../../util.h"
+#include "../soundbase.h"
+
```
Comment (hoffie) [inline src/sound/jack/sound.h:45]: Windows/Jack build reproducibly [fails](https://github.com/jamulussoftware/jamulus/runs/6544938563?check_suite_focus=true#step:9:278) due to some definition problem. Maybe this include reordering is the culprit?

File: src/util.h [hunk 2]
```diff
@@ -369,7 +375,7 @@ public slots:
     void keyPressEvent ( QKeyEvent* pEvent )
     {
         // block escape key
-        if ( pEvent->key() != Qt::Key_Escape )
+        if ( ( this->windowFlags() & Qt::Dialog ) != 0 || pEvent->key() != Qt::Key_Escape )
```
Comment (hoffie) [inline src/util.h:378]: This seems unrelated at first glance?
If it was really intended to go in, it might make sense to put it into its own commit?

**Review by hoffie — COMMENTED** [review]
>  Nothing really but I'd wait until after 3.9.0, as moving to util might not be its final resting place.

I've just tagged it for 3.9.1 based on this.
In general I'd say it should go into 3.9.0 if it is ready. We can always change it later. Otherwise, such a PR is prone to be a target for conflicts which might be annoying to resolve.
Also, 3.9.0 is currently in a state with lots of moving parts under the hood, so it should get proper "generic" testing anyway.

File: src/sound/jack/sound.h (hunk 1 above)
Comment (pljones) [reply src/sound/jack/sound.h]: `"../soundbase.h"` includes `"../../global.h"` and `"../../util.h"` (and in that order) -- this _should_ work.  Also, I can't see _any_ jack-ish dependency...  It's really weird but yeah, totally reproducible.  I'll try dropping the includes here...  Mm... better.

File: src/util.h (hunk 2 above)
Comment (pljones) [reply src/util.h:378]: Uh...  No idea how that got in there.  Hm... Good spot.  I'd been toying with the idea - it was meant to have been reverted.

Comment (pljones) [reply src/util.h:378]: Strange.  Local diff from here to master doesn't show this as a change.

Comment (pljones) [reply src/util.h:378]: OK, after rebasing I get the change.  It must have been dropped on master and my rebase didn't pick it up for some reason.

Comment (pljones) [reply src/util.h]: Hm.  And then it didn't.  I've given in and reverted to just the needed change in the sound header, which seems to make it work.  It would be good to know why the reordering is fine on Linux but not Windows.

File: src/sound/jack/sound.h (hunk 1 above)
Comment (pljones) [reply src/sound/jack/sound.h]: Hm.  And then it didn't.  I've given in and reverted to just the needed change in the sound header, which seems to make it work.  It would be good to know why the reordering is fine on Linux but not Windows.

**Review by hoffie — APPROVED** [review]

**Review by ann0see — APPROVED** [review]
As we're moving a lot of files around, I think we can merge this...

I agree with @pgScorpio that the util file gets filled with a lot of stuff...

**Comment by pljones** [comment]
> As we're moving a lot of files around, I think we can merge this...

Well, a lot of it seems to be translations - I'm not sure how they get managed so I just bundled them in as they seemed to have changed.

Merging.

**Comment by pljones** [comment]
I should have changed the milestone when merging...
